content/blog/real-time-smart-city-traffic-monitoring-using-microservices-based-streaming-architecture-part-2.md

```diff
@@ -0,0 +1,331 @@
+
+We made a single configuration change to Wildfly. We changed the port to 28080 instead of 8080, as it is already used by the sandbox. Wildfly runs in standalone mode, so the configuration file is in WILDFLY_HOME/standalone/configuration/standalone.xml.
+
+For monitoring, we let the streaming architecture work for us. We use the <a target='\_blank'  href='https://streamsets.com/products/dataops-platform/data-collector-engine/'>Streamset Data collector</a> to easily redirect the sensor data to Elasticsearch so that we can actually monitor the system with a nice dashboard with Kibana. To set up Streamsets with MapR Event Store requires some work with version 1.6, or from the <a target='\_blank'  href='https://streamsets.com/documentation/datacollector/latest/help/#Install_Config/MapR-Prerequisites.html'>official Streamsets documentation</a>).
+
+Finally, installation and setup of Elasticsearch and Kibana is <a target='\_blank'  href='https://www.digitalocean.com/community/tutorials/how-to-install-elasticsearch-logstash-and-kibana-elk-stack-on-centos-7'>well documented on Centos/RHEL</a>.
+
+For production, all those parts can be readily separated to run on separate servers. They can be run on either cluster nodes or edge nodes. If it’s a MapR cluster, installing the MapR Client and pointing it to the cluster CLDB nodes will be all the configuration needed for full access to the streams. For an Apache Kafka cluster, refer to the <a target='\_blank'  href='http://kafka.apache.org/documentation'>official Kafka documentation</a>.
+
+## Traffic Monitoring Prototype - How To
+
+**Creating the Streams with maprcli**
+
+The first task is to create streams and topics for our application. To do this, it’s a best practice to create a volume for streams first. As user mapr, type from the command line:
+
+```python
+maprcli volume create -name streams -path /streams
+maprcli stream create -path /streams/traffic -produceperm p -consumerperm p
+maprcli stream topic create -path /streams/traffic -topic data
+maprcli stream topic create -path /streams/traffic -topic agenda
+maprcli stream topic create -path /streams/traffic -topic rule-runtime
+```
+
+Note: MapR Event Store is more of a superset of Kafka than simply a clone. In addition to being faster, MapR Event Store can use all the advantages of the MapR Distributed File and Object Store, such as volumes (with permissions and quotas and so on) and replication. A cluster is not limited to just defining topics, but can define several streams which each can have several topics. Therefore, instead of a topic name, a MapR Stream has a `path:topic` notation. Here, our data stream’s full name is `“/streams/traffic:data”`.
```

Comment:
Replace "which" with "that" here
![image](https://user-images.githubusercontent.com/77496261/137516282-da448dad-64c6-4d83-aa59-a09c27fed9d0.png)

Reply:
done